src/module/gift/gift.service.ts

```diff
@@ -22,6 +22,16 @@ export class GiftService {
         return null
     }
 
+    async isGiftInStore(id: number, store: Store): Promise<boolean> {
+        const gift = await this.giftRepository.findOne({ where: { id }, relations: ['store'] })
```

Comment:
**const gift = await this.giftRepository.findOne({ where: { id, storeId: store.id }, relations: ['store'] })** ???